internal/framework/generic_resource.go

```diff
@@ -306,8 +307,8 @@ func (r *GenericResource) GetTimeout(operation string, defaultTimeout time.Durat
 
 func (r *GenericResource) AwaitReady(ctx context.Context, state ResourceState, operation string) error {
 	timeout := r.GetTimeout(operation, READINESS_TIMEOUT)
-	if timeout <= 0 {
-		// TODO(asz6): Log message saying that we are not waiting
+	if timeout == 0 {
+		tflog.Info(ctx, "Not waiting for "+r.TypeName+" to achieve desired state because "+operation+" timeout is 0")
```

Comment:
Are we going to use _structural_ logging supported by `tflog`? I.e.

```go
tflog.Info(ctx, "Not waiting to achieve desired state", "type", r.TypeName, "operation", operation, "timeout", timeout)
```

vs

```go
tflog.Info(ctx, "Not waiting for "+r.TypeName+" to achieve desired state because "+operation+" timeout is 0")
```

Reply:
I prefer the second because it makes it clearer that timeout=0 is the reason for skipping waiting, but I can incorporate that into the message and use the structured logging for the resource type and operation.

Reply:
I updated the logging statements to use structured logging. This is what the output looks like when `TF_LOG=INFO` is set:

```
$ terraform apply
...
nuodbaas_project.proj: Creating...
2024-03-05T11:13:29.123-0500 [INFO]  Starting apply for nuodbaas_project.proj
2024-03-05T11:13:30.126-0500 [INFO]  provider.terraform-provider-nuodbaas_v0.2.0: Not waiting for resource to achieve desired state because timeout is 0: operation=create tf_resource_type=nuodbaas_project tf_rpc=ApplyResourceChange tf_provider_addr=registry.terraform.io/nuodb/nuodbaas tf_req_id=03628999-d1d8-20ce-e300-ca0e373b6336 @caller=/Users/asz6/repos/terraform-provider-nuodbaas/internal/framework/generic_resource.go:311 @module=nuodbaas resourceType=project timestamp=2024-03-05T11:13:30.126-0500
nuodbaas_project.proj: Creation complete after 1s [name=proj]
nuodbaas_database.db: Creating...
2024-03-05T11:13:30.167-0500 [INFO]  Starting apply for nuodbaas_database.db
2024-03-05T11:13:30.257-0500 [INFO]  provider.terraform-provider-nuodbaas_v0.2.0: Not waiting for resource to achieve desired state because timeout is 0: @module=nuodbaas tf_req_id=4c434f99-39f1-bb43-190b-e1d062f6abf0 tf_provider_addr=registry.terraform.io/nuodb/nuodbaas tf_resource_type=nuodbaas_database tf_rpc=ApplyResourceChange @caller=/Users/asz6/repos/terraform-provider-nuodbaas/internal/framework/generic_resource.go:311 operation=create resourceType=database timestamp=2024-03-05T11:13:30.257-0500
nuodbaas_database.db: Creation complete after 0s [name=db]
...

$ terraform destroy
...
nuodbaas_database.db: Destroying... [name=db]
2024-03-05T11:15:24.376-0500 [INFO]  Starting apply for nuodbaas_database.db
2024-03-05T11:15:24.403-0500 [INFO]  provider.terraform-provider-nuodbaas_v0.2.0: Not waiting for deletion to be finalized because timeout is 0: @caller=/Users/asz6/repos/terraform-provider-nuodbaas/internal/framework/generic_resource.go:340 tf_req_id=fc0f1c65-e287-3427-c930-88dfafa9e8ed tf_resource_type=nuodbaas_database tf_rpc=ApplyResourceChange @module=nuodbaas resourceType=database tf_provider_addr=registry.terraform.io/nuodb/nuodbaas timestamp=2024-03-05T11:15:24.403-0500
nuodbaas_database.db: Destruction complete after 0s
nuodbaas_project.proj: Destroying... [name=proj]
2024-03-05T11:15:24.454-0500 [INFO]  Starting apply for nuodbaas_project.proj
2024-03-05T11:15:24.492-0500 [INFO]  provider.terraform-provider-nuodbaas_v0.2.0: Not waiting for deletion to be finalized because timeout is 0: @module=nuodbaas tf_req_id=52e10f9a-470d-7865-18d9-23d7f6cdad3d tf_resource_type=nuodbaas_project tf_rpc=ApplyResourceChange @caller=/Users/asz6/repos/terraform-provider-nuodbaas/internal/framework/generic_resource.go:340 resourceType=project tf_provider_addr=registry.terraform.io/nuodb/nuodbaas timestamp=2024-03-05T11:15:24.491-0500
nuodbaas_project.proj: Destruction complete after 0s
```